### DISCUSSION TITLE
How chakra knows which variant of a color should be used as background color, text color, etc?

### DISCUSSION THREAD

**Discussion by amirhhashemi** [chakra-ui/chakra-ui#5397] [Q&A] [answered]
I'm searching through the repo to find where the color schemes are defined but I can't find them. I can't find any documentation about how to add a new color scheme so I'm trying to figure it out myself. there should be a file that contains all color schemes (gray, blue, facebook).

EDIT: I'm not talking about the  `packages/theme/src/foundations/colors.ts` file. in this file, a list of colors and their variants are defined but at least in my head, a color scheme is a set of "rules" that defines what color a part of an element should have. if the colors.ts file contains the available colors, where are the rules are defined? when I add `colorScheme="blue"` property to an element like button, how does it know which variant of blue should be used as the background color, which one as text color, etc?

**Comment by primos63** [comment]
Well that's odd because it doesn't seem hidden.
![Chakra-colors-theme](https://user-images.githubusercontent.com/2768849/149668195-75ad4f06-3514-4cdd-9f16-f3a4444d68d2.png)
When you follow the link to the theme package, you go to src/foundations/colors.ts.

**Reply by amirhhashemi** [reply]
I know about the `colors.ts` file but I'm looking for something different (maybe because I recently started using chakra and don't know how things exactly work). the colors file obviously defines a list of colors (blue, red, etc) and their variants (blue.100, blue.200, etc) but my question is, when I add `colorScheme="blue"` property to an element like button, how it knows which variant to use as the background color, which one as text color, etc.

**Reply by primos63** [reply]
> ...my question is, when I add colorScheme="blue" property to an element like button, how it knows which variant to use as the background color, which one as text color, etc.

That's a different question from the one you originally asked. Still, I believe your questions are answered in the documentation. If you haven't you should read through the docs from top to bottom and pay special attention to the `Features` and `Theming` sections. From there you can go to a component you have a question about to read about it. There are buttons near the top of each component
![image](https://user-images.githubusercontent.com/2768849/149671260-b3081be4-cc38-4120-8577-5bc702158625.png)
which will bring you to the source code and theme section of the component. 99% of the time you are only concerned about the theme source.

**Reply by amirhhashemi** [reply]
Yes, you are right. I changed my question and I keep reading the source code. thanks for taking the time to answer me :)

**Comment by amirhhashemi** [comment, marked as answer]
Well, the answer was in front of my eyes all the time. I should read the source code more carefully.
